openupgrade_scripts/scripts/hr_holidays/17.0.1.6/pre-migration.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2024- Le Filament (https://le-filament.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ("hr.leave.stress.day", "hr.leave.mandatory.day"),
+]
+
+_table_renames = [
+    ("hr_leave_stress_day", "hr_leave_mandatory_day"),
+]
+
+
+def _hr_leave_company_id(cr):
+    """Create and set company_id value in the same order than new compute
+    (employee_company_id, mode_company_id, department_id.company_id)
+    """
+    openupgrade.logged_query(
+        cr,
+        """
+        ALTER TABLE hr_leave
+        ADD COLUMN IF NOT EXISTS company_id INTEGER;
+        """,
+    )
+    openupgrade.logged_query(
+        cr,
+        """UPDATE hr_leave AS leave
+        SET company_id = CASE
+            WHEN employee_company_id IS NOT NULL THEN employee_company_id
+            WHEN mode_company_id IS NOT NULL THEN mode_company_id
+            ELSE NULL
+            END
+        WHERE leave.company_id IS NULL""",
+    )
+    openupgrade.logged_query(
+        cr,
+        """UPDATE hr_leave AS leave
+        SET company_id = d.company_id
+        FROM hr_department AS d
+        WHERE leave.company_id IS NULL AND leave.department_id = d.id""",
+    )
+
+
+def _map_leave_accrual_level_action(cr):
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE hr_leave_accrual_level
+        SET action_with_unused_accruals = CASE
+                WHEN maximum_leave > 0 THEN 'maximum'
```

Comment:
Please fine-tune this to avoid considering it if the value is zero.
```suggestion
                WHEN maximum_leave > 0 AND postpone_max_days > 0  THEN 'maximum'
```
V16
![image](https://github.com/user-attachments/assets/7de24d09-a697-4e9c-89bb-c9faa1dba323)

V17: If the value is zero, it is better to keep the default value.
![image](https://github.com/user-attachments/assets/ff96cbf6-028f-4796-86d4-010a8baf6d8f)